pyanaconda/payload/dnf/utils.py

```diff
@@ -0,0 +1,139 @@
+    # Add /var/tmp/ if this is a directory or image installation
+    if not conf.target.is_hardware:
+        var_tmp = os.statvfs("/var/tmp")
+        structured["/var/tmp"] = Size(var_tmp.f_frsize * var_tmp.f_bfree)
+    return structured
+
+
+def pick_mount_point(df, download_size, install_size, download_only):
+    reasonable_mpoints = {
+        '/var/tmp',
+        conf.target.system_root,
+        os.path.join(conf.target.system_root, 'home'),
+        os.path.join(conf.target.system_root, 'tmp'),
+        os.path.join(conf.target.system_root, 'var'),
+    }
+
+    requested = download_size
+    requested_root = requested + install_size
+    root_mpoint = conf.target.system_root
+    log.debug('Input mount points: %s', df)
+    log.info('Estimated size: download %s & install %s', requested,
+             (requested_root - requested))
+
+    # Find sufficient mountpoint to download and install packages.
+    sufficients = {key: val for (key, val) in df.items()
+                   if ((key != root_mpoint and val > requested) or val > requested_root) and
+                   key in reasonable_mpoints}
+
+    # If no sufficient mountpoints for download and install were found and we are looking
+    # for download mountpoint only, ignore install size and try to find mountpoint just
+    # to download packages. This fallback is required when user skipped space check.
+    if not sufficients and download_only:
+        sufficients = {key: val for (key, val) in df.items() if val > requested and
+                       key in reasonable_mpoints}
+        if sufficients:
+            log.info('Sufficient mountpoint for download only found: %s', sufficients)
+    elif sufficients:
+        log.info('Sufficient mountpoints found: %s', sufficients)
+
+    if not sufficients:
+        log.debug("No sufficient mountpoints found")
+        return None
+
+    sorted_mpoints = sorted(sufficients.items(), key=operator.itemgetter(1), reverse=True)
+
+    # try to pick something else than root mountpoint for downloading
+    if download_only and len(sorted_mpoints) >= 2 and sorted_mpoints[0][0] == root_mpoint:
+        return sorted_mpoints[1][0]
+    else:
+        # default to the biggest one:
+        return sorted_mpoints[0][0]
+
+
+def do_transaction(base, queue_instance):
+    # Execute the DNF transaction and catch any errors. An error doesn't
+    # always raise a BaseException, so presence of 'quit' without a preceeding
+    # 'post' message also indicates a problem.
+    try:
+        display = PayloadRPMDisplay(queue_instance)
+        base.do_transaction(display=display)
+        exit_reason = "DNF quit"
+    except BaseException as e:  # pylint: disable=broad-except
+        log.error('The transaction process has ended abruptly')
+        log.info(e)
+        import traceback
+        exit_reason = str(e) + traceback.format_exc()
+    finally:
+        base.close()  # Always close this base.
+        queue_instance.put(('quit', str(exit_reason)))
```

Comment:
I don't think this is utils code. It's one of the main things the payload is doing not just an utility to achieve something..

Reply:
I don't think it matters, but I don't care about the name of this file. I just want a library of DNF-related functions and constants. What do you suggest?